src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -3511,6 +3513,7 @@ bool QgsPostgresProvider::changeFeatures( const QgsChangedAttributesMap &attr_ma
   {
     pushError( tr( "PostGIS error while changing attributes: %1" ).arg( e.errorMessage() ) );
     conn->rollback();
+    conn->PQexecNR( QStringLiteral( "DEALLOCATE updatefeature" ) );
```

Comment:
at line 4356 a PGException is also thrown, but there's no related PQprepare updatefature. The DEALLOCATE might perhaps not hurt, but that seems a bit messy. Wouldn't moving that line  just before throwing the exception be cleaner ?